pkg/vmprovider/providers/vsphere2/network/network.go

```diff
@@ -259,30 +260,42 @@ func createNetOPNetworkInterface(
 		return nil, fmt.Errorf("network kind %q is not supported for VDS", kind)
 	}
 
+	var err error
 	// If empty, NetOP will try to select a namespace default.
 	networkName := interfaceSpec.Network.Name
+	netIf := &netopv1alpha1.NetworkInterface{}
+	netIfKey := types.NamespacedName{
+		Namespace: vmCtx.VM.Namespace,
+		Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, true),
+	}
 
-	netIf := &netopv1alpha1.NetworkInterface{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, false),
-			Namespace: vmCtx.VM.Namespace,
-		},
+	// check if a networkIf object exists with the older (v1a1) naming convention
+	if err = client.Get(vmCtx, netIfKey, netIf); ctrlruntime.IgnoreNotFound(err) != nil {
```

Comment:
Move this on to its own line `err := client.Get(...)`, and remove the `err` declaration on L263 - I prefer that any vars in the if itself are scoped to just that.

Or with my later comment, maybe something like 
```
if err := client.Get(...); err != nil {
    if !apierrors.IsNotFound(err) {
        return err
    }

    name =  NetOPCRName(..., false)
}
```